recipes/stlab/1.7.0_and_above/conanfile.py

```diff
@@ -0,0 +1,213 @@
+
+        if self.info.settings.compiler == "gcc" and Version(self.info.settings.compiler.version) < "9":
+            raise ConanInvalidConfiguration("Need GCC >= 9")
+
+        if self.info.settings.compiler == "clang" and Version(self.info.settings.compiler.version) < "8":
+            raise ConanInvalidConfiguration("Need Clang >= 8")
+
+        if self.info.settings.compiler == "Visual Studio" and Version(self.info.settings.compiler.version) < "15.8":
+            raise ConanInvalidConfiguration("Need Visual Studio >= 2017 15.8 (MSVC 19.15)")
+
+        # Actually, we want *at least* 15.8 (MSVC 19.15), but we cannot check this for now with Conan.
+        if self.info.settings.compiler == "msvc" and Version(self.info.settings.compiler.version) < "19.15":
+            raise ConanInvalidConfiguration("Need msvc >= 19.15")
+
+        self._validate_task_system()
+        self._validate_thread_system()
+        self._validate_boost_components()
+
+    def configure(self):
+        if self.options.task_system == "auto":
+            self.options.task_system = self._default_task_system()
+
+        if self.options.thread_system == "auto":
+            self.options.thread_system = self._default_thread_system()
+
+        self.output.info("STLab Use Boost: {}.".format(self.options.use_boost))
+        self.output.info("STLab Future Coroutines: {}.".format(self.options.future_coroutines))
+        self.output.info("STLab No Standard Coroutines: {}.".format(self.options.no_std_coroutines))
+        self.output.info("STLab Task System: {}.".format(self.options.task_system))
+        self.output.info("STLab Thread System: {}.".format(self.options.thread_system))
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        tc.variables["BUILD_TESTING"] = self.options.test
+        tc.variables["STLAB_USE_BOOST_CPP17_SHIMS"] = self.options.use_boost
+        tc.variables["STLAB_NO_STD_COROUTINES"] = self.options.no_std_coroutines
+        tc.variables["STLAB_THREAD_SYSTEM"] = self.options.thread_system
+        tc.variables["STLAB_TASK_SYSTEM"] = self.options.task_system
+
+        # TODO
+        # # If main_executor == "auto" it will be detected by CMake scripts
+        # if self.options.main_executor != "auto":
+        #     tc.variables["STLAB_MAIN_EXECUTOR"] = self.options.main_executor
+
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_id(self):
+        #TODO: is header only but needs a header modified by cmake
+        # self.info.header_only()
```

Comment:
```suggestion
        self.info.settings.clear()
```

Need to check which options does not affect the generated artfacts and remove from the package ID